src/internal/easee/observationshandler.go

```diff
@@ -0,0 +1,143 @@
+package easee
+
+import (
+	"errors"
+
+	"github.com/futurehomeno/cliffhanger/adapter/service/chargepoint"
+	"github.com/futurehomeno/cliffhanger/adapter/service/numericmeter"
+
+	"github.com/futurehomeno/edge-easee-adapter/internal/signalr"
+)
+
+type ObservationsHandler interface {
+	// HandleObservation handles signalr observation callback.
+	HandleObservation(observation signalr.Observation) error
+}
+
+type observationsHandler struct {
+	chargepoints []chargepoint.Service
+	meterElecs   []numericmeter.Service
+	cache        ObservationCache
+	callbacks    map[signalr.ObservationID]func(signalr.Observation) error
+}
+
+func NewObservationsHandler(chargepoints []chargepoint.Service, meterElecs []numericmeter.Service, cache ObservationCache) ObservationsHandler {
```

Comment:
We expect the chargers to have one instance of a `chargepoint` service (same story with `numericmeter`), so there's no need to pass slices here.